spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -216,9 +219,18 @@ public Object execute(Object[] parameters) {
 							"Too few parameters are provided for query method: " + getQueryMethod().getName());
 				}
 				Object value = it.next();
-				filter.getFieldFilterBuilder().setField(fieldReference)
-						.setOp(getOperator(part, value))
-						.setValue(this.classMapper.toFirestoreValue(value));
+				StructuredQuery.FieldFilter.Builder queryFilterBuilder = filter.getFieldFilterBuilder()
+						.setField(fieldReference).setOp(getOperator(part, value));
+				if (fieldReference.getFieldPath().equals(FieldPath.documentId().toString())) {
+					FirestoreTemplate firestoreTemplate = (FirestoreTemplate) this.reactiveOperations;
+					queryFilterBuilder
+							.setValue(Value.newBuilder().setReferenceValue(
+									firestoreTemplate.buildResourceName(persistentEntity, (String) value)));
```

Comment:
For future developers, I'd add a comment for why we can assume value is always a `String`.

Reply:
Done.